crates/distribution-types/src/prioritized_distribution.rs

```diff
@@ -284,27 +286,29 @@ impl PrioritizedDist {
 }
 
 impl<'a> CompatibleDist<'a> {
-    /// Return the [`Dist`] to use during resolution.
-    pub fn for_resolution(&self) -> &Dist {
+    /// Return the [`ResolvedDistRef`] to use during resolution.
+    pub fn for_resolution(&self) -> ResolvedDistRef<'a> {
```

Comment:
This is basically the whole reason `ResolvedDistRef` exists. Maybe we should just clone here? We clone `Dist` all over the place already...

Reply:
I honestly think `ResolvedDistRef` is ok.

Reply:
It started to feel a little silly once I saw how much we clone `Dist`, but I'm okay with having it for now.